### PR TITLE
Remove some no longer useful docs

### PR DESCRIPTION
**Pull request by ahal**

### CODE REVIEW

**Comment by ahal** [comment]
I initially was just removing one page, but started seeing more and more pages that didn't need to be here. Each commit contains an explanation of why its contents are being removed.

File: gecko_tests/new_config.rst [hunk 1]
```diff
@@ -1,153 +0,0 @@
-Turning on Firefox tests for a new configuration
```
Comment (jcristau) [inline gecko_tests/new_config.rst:1]: https://firefox-source-docs.mozilla.org/testing/tests-for-new-config/index.html is currently just a pointer to this page, so we might need to update that first?

File: gecko_tests/index.rst [hunk 2]
```diff
@@ -1,16 +0,0 @@
...
-===========
-
-Gecko tests documentation
-
-.. toctree::
-    :maxdepth: 2
-    :glob:
-
-    mochitest_xpcshell_manifest_keywords
-    reftest_manifest_keywords
```
Comment (jcristau) [inline gecko_tests/index.rst:14]: I couldn't find this stuff in firefox source docs.  If it is still relevant, should it be moved there, or did I just miss it?

File: gecko_tests/new_config.rst (hunk 1 above)
Comment (ahal) [reply gecko_tests/new_config.rst:1]: Good catch! I'll port the docs over to firefox-source-docs first.

**Comment by ahal** [comment]
Here's a revision to add the Gecko testing docs to firefox-source-docs:
https://phabricator.services.mozilla.com/D187486

**Review by gbrownmozilla — APPROVED** [review]

**Pull request merged**